web/modules/custom/og_addition/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Drupal\og_addition\Plugin\EntityViewBuilder;
+
+use Drupal\Core\Url;
+use Drupal\intl_date\IntlDate;
+use Drupal\node\Entity\NodeType;
+use Drupal\node\NodeInterface;
+use Drupal\og\OgMembershipInterface;
+use Drupal\server_general\EntityDateTrait;
+use Drupal\server_general\EntityViewBuilder\NodeViewBuilderAbstract;
+use Drupal\server_general\TitleAndLabelsTrait;
+
+/**
+ * The "Node Group" plugin.
+ *
+ * @EntityViewBuilder(
+ *   id="node.group",
+ *   label=@Translation("Node - Group"),
+ *   description = "Node view builder for Group bundle."
+ * )
+ */
+class NodeGroup extends NodeViewBuilderAbstract {
+  use EntityDateTrait;
+  use TitleAndLabelsTrait;
+
+  /**
+   * Build full view mode.
+   *
+   * @param array $build
+   *   The existing build.
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array.
+   */
+  public function buildFull(array $build, NodeInterface $entity): array {
+
+    $elements = [];
+
+    $user = $this->currentUser;
+    $storage = $this->entityTypeManager->getStorage('og_membership');
```

Comment:
Please use OG's API to check if user can subscribe

Reply:
Sure, on that